File: app/src/main/java/ca/judacribz/gainzassist/models/db/SetDao.java
package ca.judacribz.gainzassist.models.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ca.judacribz.gainzassist.models.ExerciseSet;

@Dao
public interface SetDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(ExerciseSet exerciseSet);

    @Query("SELECT * from  exercise_sets")
    LiveData<List<ExerciseSet>> getAll();

    @Query("SELECT * FROM exercise_sets WHERE exercise_id = :exerciseId")
    LiveData<List<ExerciseSet>> getLiveFromExercise(long exerciseId);

    @Query("SELECT * FROM exercise_sets WHERE exercise_id = :exerciseId")
    List<ExerciseSet> getFromExercise(long exerciseId);

    @Query("SELECT * from exercise_sets WHERE id = :id")
    ExerciseSet getId(long id);

    @Update
    void update(ExerciseSet... exerciseSets);

    @Delete
    void delete(ExerciseSet... exerciseSets);
}

File: app/src/main/java/ca/gainzassist/models/db/SetDao.kt
package ca.gainzassist.models.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import ca.gainzassist.models.ExerciseSet

@Dao
interface SetDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(exerciseSet: ExerciseSet)

    @Query("SELECT * from  exercise_sets")
    fun getAll(): LiveData<List<ExerciseSet>>

    @Query("SELECT * FROM exercise_sets WHERE exercise_id = :exerciseId")
    fun getLiveFromExercise(exerciseId: Long): LiveData<List<ExerciseSet>>

    @Query("SELECT * FROM exercise_sets WHERE exercise_id = :exerciseId")
    fun getFromExercise(exerciseId: Long): List<ExerciseSet>

    @Query("SELECT * from exercise_sets WHERE id = :id")
    fun getId(id: Long): ExerciseSet?

    @Update
    fun update(vararg exerciseSets: ExerciseSet)

    @Delete
    fun delete(vararg exerciseSets: ExerciseSet)
}
